build.py

```diff
@@ -879,6 +888,19 @@ def create_dockerfile_buildbase(ddir, dockerfile_name, argmap):
         if FLAGS.enable_gpu:
             df += install_dcgm_libraries(argmap['DCGM_VERSION'],
                                          target_machine())
+        elif 'pytorch' in backends:
+            # The cpu-only build uses ubuntu as the base image, and so the cuda,
+            # openmpi, nccl and cudnn libs are not available. We must copy them from the Triton min container ourselves.
+            df += '''
+COPY --from=min_container /usr/local/cuda /usr/local/cuda
+COPY --from=min_container /usr/lib/x86_64-linux-gnu/libnccl.so.2 /usr/lib/x86_64-linux-gnu/libnccl.so.2
+COPY --from=min_container /usr/lib/x86_64-linux-gnu/libcudnn.so.8 /usr/lib/x86_64-linux-gnu/libcudnn.so.8
+
+RUN apt-get update && \
+        apt-get install -y --no-install-recommends openmpi-bin
+
+ENV LD_LIBRARY_PATH /usr/local/cuda/targets/x86_64-linux/lib:${LD_LIBRARY_PATH}
```

Comment:
This gets all the required libraries and headers into the build, but don't you need to copy the stubs and some other libraries into the final image so that Triton can actually run?

Reply:
These are only needed during runtime. Moving to final image section.